src/Backend/FeedServer.vala

```diff
@@ -56,10 +61,32 @@ public class FeedReader.FeedServer : GLib.Object {
 			Logger.debug("feedserver: engine unload %s".printf(info.get_name()));
 		});
 
+		if(Settings.general().get_string("plugin") == "none")
+		{
+			LoadAllPlugins();
+		}
+		else
+		{
+			LoadPlugin(Settings.general().get_string("plugin"));
+		}
+	}
+
+	public void LoadAllPlugins()
+	{
+		Logger.debug("FeedServer: load all available plugins");
 		foreach(var plugin in m_engine.get_plugin_list())
 		{
 			m_engine.try_load_plugin(plugin);
 		}
+
+		// have to readd this path, otherwise new icons won't show up
+		Gtk.IconTheme.get_default().add_resource_path("/org/gnome/FeedReader/icons");
```

Comment:
That's odd. Why do we need the icon path to load plugins?

Reply:
The plugins have logo icons compiled into their binaries as gresource. For some reason these icons can't be found if I don't add this path here again and only broken icons would show up on the login page.